pandas/core/indexes/numeric.py

```diff
@@ -144,15 +145,17 @@ def _ensure_array(cls, data, dtype, copy: bool):
                 data = list(data)
 
             orig = data
-            data = np.asarray(data, dtype=dtype)
+            if isinstance(data, (list, tuple)):
+                if len(data):
+                    data = sanitize_array(data, index=None)
+                else:
+                    data = np.array([], dtype=np.int64)
```

Comment:
Why int and not float?  I think in base class we do float

Reply:
My thought was that indexes default to be integer-based (to `RangeIndex`), and that ints can always be converted to floats, but not the other way around, so doing floats would be more commital.

Also it wont' matter so much, when merging this into `Index`, then `Index([]).dtype` will be object. but I can change it if you think so.

Reply:
works for me, thanks